### ISSUE TITLE
Controller panics when tag field of Image resource not in expected format.

### ISSUE THREAD

**Issue by GrahamDumpleton** [pivotal/kpack#193]
Controller crashes with:

```
panic: runtime error: index out of range [0] with length 0
--
  | goroutine 162 [running]:
  | github.com/pivotal/kpack/pkg/apis/build/v1alpha1.(*Build).Tag(...)
  | /layers/org.cloudfoundry.dep/packages/src/github.com/pivotal/kpack/pkg/apis/build/v1alpha1/build.go:43
  | github.com/pivotal/kpack/pkg/apis/build/v1alpha1.(*Build).BuildPod(0xc0005c4240, 0xc0000440a2, 0x78, 0xc000044011, 0x77, 0xc000048010, 0x86, 0xc00004412a, 0x7f, 0xc000138000, ...)
  | /layers/org.cloudfoundry.dep/packages/src/github.com/pivotal/kpack/pkg/apis/build/v1alpha1/build_pod.go:167 +0x2a84
  | github.com/pivotal/kpack/pkg/buildpod.(*Generator).Generate(0xc000340c80, 0xc0005c4240, 0x558e7518b970, 0x8, 0x558e75d3aec0)
  | /layers/org.cloudfoundry.dep/packages/src/github.com/pivotal/kpack/pkg/buildpod/generator.go:22 +0x17f
  | github.com/pivotal/kpack/pkg/reconciler/v1alpha1/build.(*Reconciler).reconcileBuildPod(0xc00033d6e0, 0xc0005c4240, 0x9, 0x0, 0x0)
  | /layers/org.cloudfoundry.dep/packages/src/github.com/pivotal/kpack/pkg/reconciler/v1alpha1/build/build.go:119 +0x13c
  | github.com/pivotal/kpack/pkg/reconciler/v1alpha1/build.(*Reconciler).Reconcile(0xc00033d6e0, 0x558e75d7c6c0, 0xc0009b2f30, 0xc00032c840, 0x35, 0xc000010e28, 0x558e75d7c6c0)
  | /layers/org.cloudfoundry.dep/packages/src/github.com/pivotal/kpack/pkg/reconciler/v1alpha1/build/build.go:89 +0x1df
  | github.com/pivotal/kpack/vendor/github.com/knative/pkg/controller.(*Impl).processNextWorkItem(0xc0003b2a00, 0x558e75d16100)
  | /layers/org.cloudfoundry.dep/packages/src/github.com/pivotal/kpack/vendor/github.com/knative/pkg/controller/controller.go:330 +0x570
  | github.com/pivotal/kpack/vendor/github.com/knative/pkg/controller.(*Impl).Run.func1(0xc0004b40d0, 0xc0003b2a00)
```

This can be triggered when format of `tag` field in `Image` isn't what is expected.

Controller then goes into `CrashLoopBackOff` until the invalid `Image` resource is deleted.

Example of value for `tag` field that appears to trigger it is:

```
apiVersion: build.pivotal.io/v1alpha1
kind: Image
metadata:
  name: sample-java-app
spec:
  tag: http://docker-registry.default.svc:5000/testing/sample-java-app
```

The use of `http://` causes the problem. The reason this was being tried was to work out how you can use an image registry that uses only `http` and not `https`. Still haven't worked out how can use an image registry over `http` as yet, so if you can indicate how that is done that would be appreciated.

**Comment by matthewmcnew** [comment]
Thanks for the catch!  https://github.com/pivotal/kpack/pull/198 will validate the `image.spec.tag` is a valid tag to prevent this from occurring. 

I commented on https://github.com/pivotal/kpack/issues/196 regarding insecure registries.

**Comment by matthewmcnew** [comment]
Closing as fixed in #198.
